### ISSUE TITLE
Length of line still checked when 'skip' attribute is being used on the function

### ISSUE THREAD

**Issue by Insomgla** [rust-lang-nursery/rustfmt#1345]
With this code : 

```
#[cfg_attr(rustfmt, rustfmt_skip)]
fn foo() -> u32{
    a_method_with_a_very_long_name_that_cannot_be_handled_by_rustfmt_even_when_skipped_and_it_is_very_wierd_but_well() + 1
}

#[cfg_attr(rustfmt, rustfmt_skip)]
fn a_method_with_a_very_long_name_that_cannot_be_handled_by_rustfmt_even_when_skipped_and_it_is_very_wierd_but_well() -> u32 {
    0u32
}

fn main() {
    let f = foo();
    println!("{}", f);
}
```

I get the following error : 
```
Rustfmt failed at foo.rs:3: line exceeded maximum length (maximum: 100, found: 122) (sorry)
Rustfmt failed at foo.rs:7: line exceeded maximum length (maximum: 100, found: 126) (sorry)
```

A more real usecase is https://github.com/rust-lang/crates.io/blob/master/src/db.rs#L61 

I can't get `rustfmt` to ignore this line.

**Comment by Insomgla** [comment]
Any input on this? Is it an actual issue or do I use rustfmt badly?

**Comment by carols10cents** [comment]
I poked around other rustfmt issues and found https://github.com/rust-lang-nursery/rustfmt/issues/1298 that says this is intentional behavior. https://github.com/rust-lang-nursery/rustfmt/issues/1080 discusses this case where an identifier is too long and rustfmt can't do anything about it.

Before finding the issue that says this is the intended behavior, I looked into changing the behavior, but it's not straightforward since [the line length check happens after visiting the nodes in the AST](https://github.com/rust-lang-nursery/rustfmt/blob/master/src/lib.rs#L491-L497), so it operates only on the text without awareness of skip attributes.
